rayon-core/src/broadcast/mod.rs

```diff
@@ -99,13 +100,22 @@ where
     OP: Fn(BroadcastContext<'_>) -> R + Sync,
     R: Send,
 {
+    let current_thread = WorkerThread::current();
+    let current_thread_addr = current_thread as usize;
+    let started = &AtomicBool::new(false);
     let f = move |injected: bool| {
         debug_assert!(injected);
+
+        // Mark as started if we are on the thread that initiated the broadcast.
+        if current_thread_addr == WorkerThread::current() as usize {
+            started.store(true, Ordering::Relaxed);
+        }
+
         BroadcastContext::with(&op)
     };
 
     let n_threads = registry.num_threads();
-    let current_thread = WorkerThread::current().as_ref();
+    let current_thread = current_thread.as_ref();
```

Comment:
preexisting, but it is very confusing that this entire function body is unsafe. Let's move to a new edition once the crate is in tree